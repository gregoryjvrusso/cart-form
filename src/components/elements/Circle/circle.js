import React from "react";
import { Container } from "./circle.styles";
import { arrayOf, bool, element, string, node, oneOfType } from "prop-types";

const Circle = ({ checked, children, color, size }) => (
  <Container data-testid="circle" checked={checked} color={color} size={size}>
    {children}
  </Container>
);

Circle.propTypes = {
  checked: bool,
  children: oneOfType([element, string, arrayOf(element), node]),
  color: string,
  size: string,
};

Circle.displayName = "Circle";

export default Circle;
