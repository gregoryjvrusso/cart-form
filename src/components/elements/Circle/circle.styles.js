import styled, { css } from "styled-components";
import Flex from "../Flex";
import { theme } from "styled-tools";

export const Container = styled(Flex)`
  align-items: center;
  background-color: transparent;
  border: 2px solid ${(props) => theme(`colors.texts.${props.color}`)};
  border-radius: 50%;
  height: ${(props) => props.size}px;
  justify-content: center;
  width: ${(props) => props.size}px;
  ${(props) =>
    props.checked &&
    css`
      background: ${(props) => theme(`colors.texts.${props.color}`)};
    `}
`;
