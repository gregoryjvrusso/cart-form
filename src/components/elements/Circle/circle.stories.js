import React from "react";
import { storiesOf } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import theme from "../../../common/theme";
import { check } from "../../../assets/index";

import Circle from "../Circle";

storiesOf("Elements/Circle", module)
  .add("default", () => (
    <ThemeProvider theme={theme}>
      <Circle size="22" color="red">
        1
      </Circle>
    </ThemeProvider>
  ))
  .add("checked", () => (
    <ThemeProvider theme={theme}>
      <Circle size="22" color="red" checked>
        <img src={check} alt={"checked"} />
      </Circle>
    </ThemeProvider>
  ))
  .add("black default", () => (
    <ThemeProvider theme={theme}>
      <Circle size="22" color="black">
        1
      </Circle>
    </ThemeProvider>
  ))
  .add("black checked", () => (
    <ThemeProvider theme={theme}>
      <Circle size="22" color="black" checked>
        <img src={check} alt={"checked"} />
      </Circle>
    </ThemeProvider>
  ));
