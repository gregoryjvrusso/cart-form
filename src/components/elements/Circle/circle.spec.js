import React from "react";
import { render } from "@testing-library/react";
import Circle from "../Circle";

describe("Circle component", () => {
  it("deve conseguir renderizar [snapshot]", async () => {
    const { container } = render(
      <Circle size="22" color="red">
        1
      </Circle>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("verificar se os valores do css estÃ£o corretos", () => {
    const { getByTestId } = render(
      <Circle size="22" color="red">
        1
      </Circle>
    );
    expect(getByTestId("circle")).toHaveStyle("height: 22px");
    expect(getByTestId("circle")).toHaveStyle("color: red");
  });
});
