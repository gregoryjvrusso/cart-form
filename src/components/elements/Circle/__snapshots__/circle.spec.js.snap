// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component deve conseguir renderizar [snapshot] 1`] = `
<div
  class="sc-AxjAm caeAtN sc-AxirZ iWzAtR sc-AxiKw cDKsYq"
  color="red"
  data-testid="circle"
  size="22"
>
  1
</div>
`;
